FreeRTOS-Plus/Test/CBMC/proofs/Socket/vSocketBind/ALLOW_SOCKET_SEND_WITHOUT_BIND/vSocketBind_harness.c

```diff
@@ -0,0 +1,69 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "list.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_Sockets.h"
+
+
+void *safeMalloc(size_t xWantedSize) {
+        return nondet_bool() ? malloc(xWantedSize) : NULL;
+}
+
+uint16_t prvGetPrivatePortNumber( BaseType_t xProtocol )
+{
+	uint16_t usResult;
+	return usResult;
+}
+
+BaseType_t xIPIsNetworkTaskReady( void )
+{
+	/* Return true saying that the task is ready. */
+	return pdTRUE;
+}
+
+
+/* Random number generator provided by the application. In our case, CBMC provides
+ * an indeterministic value. */
+BaseType_t xApplicationGetRandomNumber( uint32_t *pulNumber )
+{
+	BaseType_t xResult;
+	return xResult;
+}
+
+/* Memory assignment for FreeRTOS_Socket_t */
+FreeRTOS_Socket_t * ensure_FreeRTOS_Socket_t_is_allocated () {
+	FreeRTOS_Socket_t *pxSocket = malloc(sizeof(FreeRTOS_Socket_t));
+
+	pxSocket->u.xTCP.rxStream = malloc(sizeof(StreamBuffer_t));
+	pxSocket->u.xTCP.txStream = malloc(sizeof(StreamBuffer_t));
+	pxSocket->u.xTCP.pxPeerSocket = malloc(sizeof(FreeRTOS_Socket_t));
+
+	return pxSocket;
+}
```

Comment:
Are these proofs just testing two different configurations?  There is a method of describing configurations you might want to use, in case in the future there are even more you want to check.

Reply:
Yes it is doing that only. I do not know the method of describing configurations. Can you point me to a reference?